src/otx/algo/instance_segmentation/maskdino.py

```diff
@@ -0,0 +1,537 @@
+            torch.nn.BatchNorm2d,
+            torch.nn.BatchNorm3d,
+            torch.nn.SyncBatchNorm,
+            torch.nn.GroupNorm,
+            torch.nn.InstanceNorm1d,
+            torch.nn.InstanceNorm2d,
+            torch.nn.InstanceNorm3d,
+            torch.nn.LayerNorm,
+            torch.nn.LocalResponseNorm,
+        )
+
+        params = []
+        uniques = set()
+        for module_name, module in model.named_modules():
+            for module_param_name, value in module.named_parameters(recurse=False):
+                if not value.requires_grad:
+                    continue
+                # Avoid duplicating parameters
+                if value in uniques:
+                    continue
+                uniques.add(value)
+
+                hyperparams = copy.copy(defaults)
+                if "backbone" in module_name:
+                    hyperparams["lr"] = hyperparams["lr"] * backbone_multiplier
+                if "relative_position_bias_table" in module_param_name or "absolute_pos_embed" in module_param_name:
+                    hyperparams["weight_decay"] = 0.0
+                if isinstance(module, norm_module_types):
+                    hyperparams["weight_decay"] = weight_decay_norm
+                if isinstance(module, torch.nn.Embedding):
+                    hyperparams["weight_decay"] = weight_decay_embed
+                params.append({"params": [value], **hyperparams})
+        return params
+
+    @staticmethod
+    def _add_grad_clipping(
+        optim: torch.optim.Optimizer,
+        clip_gradient_value: float = 0.01,
+    ) -> torch.optim.Optimizer:
+        """Add gradient clipping to the optimizer."""
+
+        class GradientClippingOptimizer(optim.__class__):
+            def step(self, closure: Callable | None = None) -> None:
+                all_params = itertools.chain(*[x["params"] for x in self.param_groups])
+                torch.nn.utils.clip_grad_norm_(all_params, clip_gradient_value)
+                super().step(closure=closure)
+
+        return GradientClippingOptimizer
+
+    def _customize_outputs(
+        self,
+        outputs: dict[str, Tensor],  # type: ignore[override]
+        inputs: InstanceSegBatchDataEntity,
+    ) -> OTXBatchLossEntity | InstanceSegBatchPredEntity:
+        if self.training:
+            return sum(outputs.values())
+
+        masks, bboxes, labels, scores = self.post_process_instance_segmentation(
+            outputs,
+            inputs.imgs_info,
+        )
+
+        if self.explain_mode:
+            msg = "Explain mode is not supported yet."
+            raise NotImplementedError(msg)
+
+        return InstanceSegBatchPredEntity(
+            batch_size=inputs.batch_size,
+            images=inputs.images,
+            imgs_info=inputs.imgs_info,
+            scores=scores,
+            bboxes=bboxes,
+            masks=masks,
+            polygons=[],
+            labels=labels,
+        )
+
+    def post_process_instance_segmentation(
```

Comment:
What about inserting this in head?
I think lit module is inappropriate to do this operation and the model should support this.

Reply:
I think this part still needs further optimization and fine-tuning for both PyTorch and OpenVINO export. I'm currently experimenting with fine-tuning the post-processing. In short, refactoring and optimization are in progress, and changes are expected in the next phase, so I’ll leave it as is for now.